packages/elasticsearch-asset-apis/src/elasticsearch-reader-api/ElasticsearchReaderAPI.ts

```diff
@@ -111,21 +112,100 @@ export class ElasticsearchReaderAPI {
     /**
      * Fetch a given slice, the data will be returned in the format
      * specified under `response_type` in the config
-    */
+     *
+     * `fetch` now sets the query `size` parameter to 1.5 times the count
+     * recorded in the slice, rather than the window size of the index.
+     * It will retry up to five times if the size of the result set comes back
+     * the same size as the querySize, this is meant to ensure all of the
+     * records are retrieved, but should be unlikely to happen in most cases
+     *
+     * When it previously used the indices max window size, it appeared to
+     * slow ES queries down significantly, refs:
+     *   https://github.com/terascope/elasticsearch-assets/issues/948
+     */
     async fetch(queryParams: ReaderSlice = {}): Promise<DataEntity[]|DataFrame|Buffer> {
-        // attempt to get window if not set
+        const countExpansionFactor = 1.5;
+        let querySize = 10000;
+        const retryLimit = 5;
```

Comment:
Note the values selected for these constants/defaults are fairly arbitrary.  I don't want to make `countExpansionFactor` too big because I don't want to hit the index.max_window_size.  Given that retries are pretty unlikely, I only let it retry a relatively small number, 5 times.  The `querySize` default is a total guess, I don't even know when this value will get used.  When there is no `count` on a slice?